src/App.tsx

```diff
@@ -15,49 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+type Goods = string[];
+
+type Props = {
+  sortField: string;
+  isSorted: boolean;
+};
+
+function getPreparedGoods(
+  goods: Goods,
+  { sortField, isSorted }: Props,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isSorted) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [isSorted, setIsSorted] = useState<boolean>(false);
```

Comment:
it's redundant to specify primitives types
```suggestion
  const [sortField, setSortField] = useState('');
  const [isSorted, setIsSorted] = useState(false);
```

Reply:
not fixed